src/runtime/profiler_common.cpp

```diff
@@ -624,12 +679,16 @@ __attribute__((destructor))
 #endif
 WEAK void
 halide_profiler_shutdown() {
+    using namespace Halide::Runtime::Internal::Synchronization;
+
     halide_profiler_state *s = halide_profiler_get_state();
     if (!s->sampling_thread) {
         return;
     }
 
-    s->current_func = halide_profiler_please_stop;
+    int one = 1;
+    atomic_store_relaxed(&(s->shutdown), &one);
```

Comment:
What is the purpose of having a relaxed "atomic" operation and this auxiliary function? Doesn't the C++ memory consistency model guarantee the same behavior even without this relaxed atomic store on all platforms?

Reply:
The implementation of atomic_store_relaxed is just *a = *b, so this is really just documentation that this is an intentional unlocked communication of a value across threads.